homework7/hw7_task1/task_find_occurrences/find_occurrences.py

```diff
@@ -0,0 +1,33 @@
+"""
+Given a dictionary (tree), that can contains multiple nested structures.
+Write a function, that takes element and finds the number of occurrences
+of this element in the tree.
+Tree can only contains basic structures like:
+    str, list, tuple, dict, set, int, bool
+"""
+from typing import Any, Iterable, Iterator, Mapping
+
+
+def recursively_read(tree: Iterable, element, occurrences: list):
+    iterator: Iterator
+    if isinstance(tree, Mapping):
+        iterator = iter(tree.values())
+    else:
+        iterator = iter(tree)
+    while True:
```

Comment:
It is a very common pattern to iterate over objects in that way (while - next() - StopIteration).   It even has a special syntax for it.   Let's use it, this will make code cleaner and reduce cognitive complexity a lot.

Reply:
It's better, but I am talking about a different approach.  Probably it will be 11 too.

Reply:
I made it! Cognitive complexity now is 9, cool! Thanks